tasks/system/Linux.yml

```diff
@@ -1,117 +1,129 @@
     set_fact:
-      sonar_installation: "{{ sonar_path }}/{{ sonar_installation_result.files.0 }}"
-      distro: "{{ ansible_facts.system | lower }}"
+      sonar_installation: '{{ sonar_path }}/{{ sonar_installation_result.files.0 }}'
       arch: "{{ 'x86-32' if ansible_architecture == 'i386' else 'x86-64' }}"
-    when: sonar_installation_result.files is defined
 
-    # continuation of ansible 2.6.* workaround
-  - name: "SonarQube installation facts"
-    set_fact:
-      sonar_installation: "{{ sonar_path }}/\
-      sonarqube-{{ sonar_major_version }}.{{ sonar_minor_version }}/"
-      distro: "{{ ansible_facts.system | lower }}"
-      arch: "{{ 'x86-32' if ansible_architecture == 'i386' else 'x86-64' }}"
-    when: sonar_installation_result.files is not defined
-
-  - name: "Render sonar.properties"
+  - name: Render sonar.properties
     template:
-      src: "sonar.properties.j2"
-      dest: "{{ sonar_installation }}conf/sonar.properties"
-      mode: "0400"
-      owner: "{{ sonar_user }}"
+      src: sonar.properties.j2
+      dest: '{{ sonar_installation }}conf/sonar.properties'
+      mode: 0400
+      owner: '{{ sonar_user }}'
 
-  - name: "Render sonarqube.service"
+  - name: Render sonarqube.service
     template:
-      src: "sonarqube.service.j2"
-      dest: "/etc/systemd/system/sonarqube.service"
-      mode: "0444"
-      owner: "{{ sonar_user }}"
+      src: sonarqube.service.j2
+      dest: /etc/systemd/system/sonarqube.service
+      mode: 0444
+      owner: '{{ sonar_user }}'
 
-  - name: "Systemd daemon-reload"
+  - name: Systemd daemon-reload
     systemd:
       daemon_reload: True
 
-  - name: "Enable sonarqube.service"
-    service:
-      name: "sonarqube"
-      enabled: True
-
-  - name: "Check nginx conf directory"
+  - name: Check nginx conf directory
     stat:
       path: /etc/nginx/conf.d
     register: nginx_conf_dir
 
-  - name: "Render nginx config"
+  - name: Render nginx config
     template:
-      src: "nginx.sonar.conf.j2"
-      dest: "/etc/nginx/conf.d/sonar.conf"
-      mode: "0444"
-      owner: "{{ sonar_user }}"
-    when: sonar_proxy_type == "nginx" and nginx_conf_dir.stat.exists and nginx_conf_dir.stat.isdir
+      src: nginx.sonar.conf.j2
+      dest: /etc/nginx/conf.d/sonar.conf
+      mode: 0444
+      owner: '{{ sonar_user }}'
+    when:
+      - sonar_proxy_type == 'nginx'
+      - nginx_conf_dir.stat.exists
+      - nginx_conf_dir.stat.isdir
+
+  - name: Download plugins
+    get_url:
+      url: '{{ item }}'
+      dest: '{{ sonar_installation }}extensions/plugins'
+      mode: 0644
+      owner: '{{ sonar_user }}'
+      group: '{{ sonar_group }}'
+    register: download_plugins_result
+    until: download_plugins_result is succeeded
+    retries: 3
+    delay: 5
+    with_items:
+      - '{{ sonar_plugins }} + {{ sonar_optional_plugins }}'
+
+  - name: Restart sonarqube
+    service:
+      name: sonarqube
+      state: restarted
+      enabled: True
```

Comment:
Should we replace with?
- name: Restart sonarqube
  systemd:
    state: restarted
    daemon_reload: yes
    enabled: True
    name: sonarqube

Reply:
Done, thanks.